botocore/auth.py

```diff
@@ -344,14 +341,16 @@ def _set_necessary_date_headers(self, request):
         # header.  Otherwise we use the X-Amz-Date header.
         if 'Date' in request.headers:
             del request.headers['Date']
+            datetime_timestamp = datetime.datetime.strptime(
+                request.context['timestamp'], '%Y%m%dT%H%M%SZ')
```

Comment:
@danielgtaylor I just did it to avoid bloating the ``context`` dictionary. The current timestamp is of the form ``'YYYYMMDDHHMMSSZ'``, which is what is the form most used in the signing code. Since this form is used once in the entire sigv4 signing process, I figured it was not worth adding a new entry to the context dictionary. Let me know what you think.